crates/services/p2p/src/peer_manager.rs

```diff
@@ -611,10 +603,9 @@ impl PeerManager {
         if !self.reserved_peers.contains(&peer_id) {
             // check if all the slots are already taken
             if non_reserved_peers_connected >= self.max_non_reserved_peers {
-                // Too many peers already connected, disconnect the Peer
-                self.pending_events.push_back(PeerInfoEvent::TooManyPeers {
+                // Too many peers already connected, disconnect the Peer with the first priority.
+                self.pending_events.push_front(PeerInfoEvent::TooManyPeers {
```

Comment:
I think we need to disconnect from peers in the first place. Otherwise, we will have a pending task that will receive incoming messages that we plan to drop.